libraries/AP_NavEKF2/AP_NavEKF2_MagFusion.cpp

```diff
@@ -1174,3 +1213,111 @@ void NavEKF2_core::recordMagReset()
     yawInnovAtLastMagReset = innovYaw;
 }
 
+// Reset states using yaw from EKF-GSF and velocity and position from GPS
+bool NavEKF2_core::EKFGSF_resetMainFilterYaw()
+{
+    // Don't do a reset unless permitted by the EK3_GSF_USE and EKF3_GSF_RUN parameter masks
+    if ((yawEstimator == nullptr)
+        || !(frontend->_gsfUseMask & (1U<<core_index))
+        || EKFGSF_yaw_reset_count >= frontend->_gsfResetMaxCount) {
+        return false;
+    };
+
+    float yawEKFGSF, yawVarianceEKFGSF;
+    if (yawEstimator->getYawData(&yawEKFGSF, &yawVarianceEKFGSF) && is_positive(yawVarianceEKFGSF) && yawVarianceEKFGSF < sq(radians(15.0f))) {
+
+        // keep roll and pitch and reset yaw
+        resetQuatStateYawOnly(yawEKFGSF, yawVarianceEKFGSF, false);
+
+        // record the emergency reset event
+        EKFGSF_yaw_reset_request_ms = 0;
+        EKFGSF_yaw_reset_ms = imuSampleTime_ms;
+        EKFGSF_yaw_reset_count++;
+
+        if (!use_compass()) {
+            gcs().send_text(MAV_SEVERITY_INFO, "EKF3 IMU%u yaw aligned using GPS",(unsigned)imu_index);
+        } else {
+            gcs().send_text(MAV_SEVERITY_WARNING, "EKF3 IMU%u emergency yaw reset - mag sensor stopped",(unsigned)imu_index);
```

Comment:
should be EKF2

Reply:
I've fixed this in the follow-up commits at the end of the branch linked in the comments.